### ISSUE TITLE
Spawning or stopping a container `sync`'s the entire filesystem hosting `/var/lib/docker`

### ISSUE THREAD

**Issue by Goirad** [moby/moby#47630]
### Description

There is an (AFAICT) undocumented cost to spawning a container, which is that `/var/lib/docker`'s filesystem is `sync`d. Depending on the host environment, this is an egregious price to pay. In the worst case, it can take minutes to flush all dirty pages to disk.

For example, if you use containers on a large machine with a TB of RAM, and you use those containers to do a lot disk writes, you can easily end up with >30 GiB of dirty pages in the page cache. This is not a problem usually, that's what the RAM is for. But if you _also_ want to spawn an alpine container on that machine to `echo "hello world"`, that spawn will block until all of those 30 GiB or dirty pages are flushed to disk.

Thanks to https://github.com/moby/moby/pull/42349#issuecomment-841753687 and the linked blog, it's because overlayfs calls `sync_filesystem` on the upper directory, which in most cases lives in `/var/lib/docker`. And because of the init layer, this happens on unmounting the init layer's overlayfs, _before_ the main container can spin up.

The documentation strongly recommends volumes for write heavy workloads, but _in general_ that is not enough to prevent this problem, since volume _also_ write data to `/var/lib/docker/`.

Of course there are workarounds for this, like having a dedicated data partition and associated filesystem, that gets mounted into the write heavy containers. But as far as I can tell this "different filesystem" point is not written down anywhere.

I also know that the docker team is not responsible for overlayfs' behavior, but it'd be nice if this could be called out somewhere (even if just in this issue).

TLDR: Write heavy workloads can cause container spawn/stop operations to take extremely long amount of time to complete. This is because unmounting an overlayfs causes a `sync` of the entire filesystem owning `/var/lib/docker`.

### Reproduce

Create some dirty pages:
```
docker run -it --rm alpine  -c 'while true; do h head -c $((1024 * 1024)) /dev/zero >> foo; sleep 1; done'
```
Observe the dirty pages growing on your system:
```
while true; do cat /proc/meminfo  | grep irty; sleep 1; done
```
Try to spawn a container:
```
docker run -it --rm alpine echo hello
```
Observe the dirty pages get flushed to disk.

Unfortunately the `sync` is internal to the `umount2` syscall, otherwise I'd have used strace to demonstrate it happening.

### Expected behavior

Strictly speaking, this is not an issue with Docker. So the expected thing is that this be documented somewhere since overlayfs is the recommended driver.

### docker version

```bash
Client: Docker Engine - Community
 Version:           25.0.5
 API version:       1.44
 Go version:        go1.21.8
 Git commit:        5dc9bcc
 Built:             Tue Mar 19 15:05:20 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          25.0.5
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       e63daec
  Built:            Tue Mar 19 15:05:20 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client: Docker Engine - Community
 Version:    25.0.5
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.13.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.25.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 9
  Running: 0
  Paused: 0
  Stopped: 9
 Images: 69
 Server Version: 25.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
 Kernel Version: 5.15.0-94-generic
 Operating System: Ubuntu 20.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 23.23GiB
 Name: dgt480s
 ID: COSP:RR3P:ZE3E:FYLT:IJ3P:GXGK:P5E4:QG3E:IKZH:QJMC:Z4FX:LEEP
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```
